apps/emqx_resource/src/emqx_resource.erl

```diff
@@ -309,7 +311,7 @@ set_resource_status_connecting(ResId) ->
 -spec get_instance(resource_id()) ->
     {ok, resource_group(), resource_data()} | {error, Reason :: term()}.
 get_instance(ResId) ->
-    emqx_resource_manager:lookup(ResId).
+    emqx_resource_manager:ets_lookup(ResId, [metrics]).
```

Comment:
Personally, I prefer add a new function for API call, the `emqx_resource:lookup` not only used by HTTP API

Reply:
In general, I agree with your consideration, but in this case I think changing the existing function is reasonable.
1. All users of this function are either API handler related code or tests. API related code should benefit from this function no longer being blocking, and tolerate the slight loss of linearizability (e.g. read the state of resource which is in the process of removal).
2. Aforementioned loss is practically non-existent since the cache now is kept consistent with the state of a resource.